src/ports/postgres/modules/graph/measures.py_in

```diff
@@ -184,10 +184,7 @@ class Graph(object):
                 -- Filtering 'Infinity' occurs in CASE instead of WHERE clause
                 -- so that the edge is part of the average value i.e. part of
                 -- the count of paths but zero addition to sum of distances.
-                AVG(CASE WHEN {e.weight} = 'Infinity'::double precision
-                            THEN 0::double precision
-                            ELSE {e.weight}::double precision
-                    END) as avg_path_length
+                AVG({e.weight}::double precision) as avg_path_length
```

Comment:
We should remove the comment above, which still references the CASE block that's no longer there.

Also wondering:

As I understand it, the original author thought it was important for infinite paths to be included in the denominator of the average value calculation (avg = sum(weights) / count(weights) ) but not in the numerator.  That doesn't seem right to me, but the fact they went to the trouble of specifically pointing it out in the comment makes me wonder if there was some reason for it that we're missing. 

Since we are changing the way in which average path length is calculated, affecting the result that gets returned (not just omitting infinite rows from the table), I just want to make sure we're doing this intentionally.  The average path length returned previously would have been a smaller number than it will be now for disconnected graphs, right?

Reply:
@njayaram2 Wondering if you recall this and and can provide some guidance here?  thanks

Reply:
It seems the average path length is usually defined for connected graphs. If a graph is unconnected, the average path length becomes infinity as well. There are suggestions like using the largest connected component's avg path length (which is possible to do with MADlib wcc function) [https://mathinsight.org/definition/network_mean_path_length](url).

I think the previous way of calculating was plain wrong. With this commit, the average length of non-infinite paths is calculated, which may or may not be useful. If we want to specifically limit this function to connected graphs, we can do that as well.

Reply:
Sounds good to me!

Also backing up this decision, I just noticed this:
```
postgres=# select madlib.graph_avg_path_length();
                             graph_avg_path_length
-------------------------------------------------------------------------------

 This function computes the average of the shortest paths between each pair of
 vertices. Average path length is based on "reachable target vertices", so it
 ignores infinite-length paths between vertices that are not connected.

 For more details:
     SELECT madlib.graph_avg_path_length('usage')

(1 row)
```
Pretty clear the previous behavior was a bug!  Glad we've fixed it.

Reply:
If you have, say, n disconnected sub-graphs of roughly the same size, this calc is still OK, right?  i.e., it just drops vertex pairs that are not connected (infinite dist) and would calculate average path length based on pairwise distances that are not infinite.

Reply:
Yes. The number and/or size of components are irrelevant, we just look at every non-infinite path and take the average.

Reply:
I see.  In that case I think this PR is ready to merge.  Thanks @reductionista for asking an interesting question to start, and to both or you for sorting it out.